lib/node_modules/@stdlib/array/base/cusome/lib/assign.js

```diff
@@ -0,0 +1,312 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2024 The Stdlib Authors.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+'use strict';
+
+// MODULES //
+
+var isComplex128Array = require( '@stdlib/array/base/assert/is-complex128array' );
+var isComplex64Array = require( '@stdlib/array/base/assert/is-complex64array' );
+var isBooleanArray = require( '@stdlib/array/base/assert/is-booleanarray' );
+var arraylike2object = require( '@stdlib/array/base/arraylike2object' );
+var reinterpret128 = require( '@stdlib/strided/base/reinterpret-complex128' );
+var reinterpret64 = require( '@stdlib/strided/base/reinterpret-complex64' );
+var reinterpretBoolean = require( '@stdlib/strided/base/reinterpret-boolean' );
+
+
+// FUNCTIONS //
+
+/**
+* Cumulatively tests whether at least n array elements in a provided array are truthy.
+*
+* @private
+* @param {Collection} x - input array
+* @param {integer} n - number of elements to be truthy
+* @param {Collection} y - output array
+* @param {integer} stride - output array stride
+* @param {NonNegativeInteger} offset - output array offset
+* @returns {Collection} output array
+*
+* @example
+* var x = [ false, false, false, true, true ];
+* var n=2;
+* var y =  [ false, null, false, null, false, null, false, null, false, null ];
+*
+* var out = indexed( x, n, y, 2, 0 );
+* // returns [ false, null, false, null, false, null, false, null, true, null ]
+*/
+function indexed( x, n, y, stride, offset ) {
+	var flg;
+	var io;
+	var i;
+
+	flg = false;
+	io = offset;
+	for ( i = 0; i < x.length; i++ ) {
+		if ( flg === false && x[ i ] && n > 0 ) {
+			n -= 1;
+		}
+		if (n === 0) {
+			flg = true;
+		}
+		y[ io ] = flg;
+		io += stride;
+	}
+	return y;
+}
+
+/**
+* Cumulatively tests whether at least n array elements in accessor array are truthy.
```

Comment:
```suggestion
* Cumulatively tests whether at least `n` array elements in accessor array are truthy.
```

Backticks in all descriptions in this file.